### DISCUSSION TITLE
triggering a mission when clicking on spaceport while just ended a mission

### DISCUSSION THREAD

**Discussion by daggs1** [endless-sky/endless-sky#9582] [General]
Greetings,

I've encountered this several times in the game where I land on a planet from a mission, get a msg asking to meet in the spaceport, click the spaceport (without the need to depart) and getting to the next mission, I'm trying to do the same but it seems I always need to depart to get it to work, there are my two missions:

> mission "test: mission 7"
> 	name "test: mission 7"
> 	description "test: mission 7"
> 	source "Ssil Vida"
> 
> 	to offer
> 		has "test: mission 6: declined"
> 
> 	on offer
> 		conversation
> 			`test: mission 7`
> 		fail
> 
> mission "test: mission 8"
> 	name "test: mission 8"
> 	description "test: mission 8"
> 	source "Ssil Vida"
> 	destination Mnai
> 
> 	to offer
> 		has "test: mission 7: declined"
> 
> 	on offer
> 		conversation
> 			`test: mission 8`
> 				accept
> 
> 	on complete
> 		dialog `test: mission 8"`

any idea what can be the issue?

**Comment by mOctave** [comment]
The standard procedure for this kind of thing is as follows:

```
mission "mission 1"
	name "mission 1"
	destination "Some planet"
	on complete
		conversation
			`When you land on some planet, you are approached by the theoretical people you just carried here from <origin>. As they obviously don't exist in this mission definition, you seem to be going insane. They offer to meet you in the spaceport in some arbitrary length of time that the game doesn't care about anyways.`

mission "mission 2"
	name "mission 2"
	source "Some planet"
	to offer
		has "mission 1: done"
	on offer
		conversation
			`This mission triggers as soon as you enter the spaceport "bar." Your fictitious passengers are still nowhere to be seen, but they offer to pay you <payment> in exchange for transport to somewhere across the galaxy.`
			choice
				`Yes, sure, whatever, I'll do it`
					accept
				`You want transport to <destination>! That's all the way across the galaxy. No way am I taking you there.`
					decline
```

Obviously, I'm leaving stuff out, but that's the general format: an `on complete` dialog offered at the end of a mission followed by a `on offer` dialog in the spaceport.

The way you're doing it _could_ also work, but keep in mind that the attributes _don't get set until you take off/land_. That's why you were having to take off in your example.

If you want to use what you put above, then you'll want a `landing` attribute like so:
```
mission "test: mission 7"
	landing
	name "test: mission 7"
```

... and on mission 8 you'll still be checking for `test: mission 6: declined`, since mission 7 hasn't been set yet. Don't add a `landing` attribute to mission 8, and it will offer in the spaceport, after you are already shown mission 7.

**Comment by daggs1** [comment]
thanks, it worked
